### DISCUSSION TITLE
Mock Service Worker Addon

### DISCUSSION THREAD

**Discussion by MatthewThornton** [storybookjs/storybook#25049] [Help]
### Summary

I am trying to get Mock Service Worker working with Storybook. 
I followed the following steps: https://storybook.js.org/addons/msw-storybook-addon 

When I run storybook I get the following error. 
` Package path ./native is not exported from package ... node_modules/msw` 

The module is there. I can see the export in the msw package json. 

### Additional information

Full error 
```
99% end closing watch compilationWARN Force closed preview build
ModuleNotFoundError: Module not found: Error: Package path ./native is not exported from package /Users/mattthornton/Repos/mvba-suit/src/SuitSystem.UI/node_modules/msw (see exports field in /Users/mattthornton/Repos/mvba-suit/src/SuitSystem.UI/node_modules/msw/package.json)
```

This is my `preview.ts` file. 

```

import { withRouter } from 'storybook-addon-react-router-v6';
import '../src/stylesheets/main.scss';
import { initialize, mswLoader } from 'msw-storybook-addon';
import { Preview } from '@storybook/react';

// Initialize MSW
initialize();

const preview: Preview = {
    parameters: {
        actions: { argTypesRegex: '^on[A-Z].*' },
        },
        reactRouter: {
            routePath: '/'
        }
    },
    loaders: [mswLoader],
    decorators: [
        withRouter,
    ]
};
export default preview;


```

This is my main.ts file 

```
import type { StorybookConfig } from '@storybook/react-webpack5';

const config: StorybookConfig = {
    stories: [
        '../src/**/*.stories.mdx',
        '../src/**/*.stories.@(js|jsx|ts|tsx)'
    ],

    addons: [
        '@storybook/addon-links',
        '@storybook/addon-essentials',
        '@storybook/addon-interactions',
        '@storybook/preset-create-react-app',
        '@storybook/addon-mdx-gfm',
        'msw-storybook-addon'
    ],

    framework: {
        name: '@storybook/react-webpack5',
        options: {}
    },

    docs: {
        autodocs: true
    },
    staticDirs: ['../public'], //👈 Configures the static asset folder in Storybook
};
export default config;
```

Versions 
`   "storybook": "^7.4.6",` 
`  "msw": "^2.0.9",`
`    "msw-storybook-addon": "^1.10.0",`

**Comment by MatthewThornton** [comment]
Changing the add on to this version fixed the issue. 
`npm install msw-storybook-addon@2.0.0--canary.122.b3ed3b1.0`

**Reply by bozanbozkurt** [reply]
Worked for me too, thanks.

**Reply by cfollet** [reply]
Just for your information (and if it saves from testing), even if `2.0.0-beta.0` in more recent than `2.0.0--canary.122.b3ed3b1.0`, this version doesn't include the fix for ` Package path ./native is not exported from package ... node_modules/msw`

**Comment by jonniebigodes** [comment]
@MatthewThornton, thanks for reaching out to us with your question. We appreciate it. Last month, The MSW release introduced some changes that the addon needs to account for. So that you and everyone else are aware and want to check on the progress of the update for the addon, it's being tracked [here](https://github.com/mswjs/msw-storybook-addon/issues/121), and as you pointed out, we have some pre-release versions out. 

Nonetheless, thanks for trying it out and confirming that it's working for your environment.
